wgpu/src/renderer.rs

```diff
@@ -137,17 +131,16 @@ impl Renderer {
 
     fn draw_primitive<'a>(
         &mut self,
+        translation: Vector,
```

Comment:
It turns out that most of the plumbing was already there.

I basically removed the `offset` field from `Layer` and added an explicit translation for each `Primitive`. This allows us to control the translation of children without worrying about unintended side-effects.